ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            fclose_function(pf);
+            return 1;
+        }
+
+        if (pMMBDXP->pField[nIField].FieldType == 'F')
+            pMMBDXP->pField[nIField].FieldType = 'N';
+
+        pMMBDXP->pField[nIField]
+            .FieldName[MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF - 1] = '\0';
+        if (EQUAL(pMMBDXP->pField[nIField].FieldName,
+                  szMMNomCampIdGraficDefecte))
+            pMMBDXP->IdGraficField = nIField;
+
+        if (pMMBDXP->pField[nIField].BytesPerField == 0)
+        {
+            if (!MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+            {
+                free(pMMBDXP->pField);
+                fclose_function(pf);
+                return 1;
+            }
+            if (pMMBDXP->pField[nIField].FieldType != 'C')
+            {
+                free(pMMBDXP->pField);
+                fclose_function(pf);
+                return 1;
+            }
+
+            memcpy(&pMMBDXP->pField[nIField].BytesPerField,
+                   (char *)(&pMMBDXP->pField[nIField].reserved_2) + 3,
+                   sizeof(MM_BYTES_PER_FIELD_TYPE_DBF));
+        }
+
+        if (nIField)
+            pMMBDXP->pField[nIField].AcumulatedBytes =
+                (pMMBDXP->pField[nIField - 1].AcumulatedBytes +
+                 pMMBDXP->pField[nIField - 1].BytesPerField);
+        else
+            pMMBDXP->pField[nIField].AcumulatedBytes = 1;
+
+        if (pszRelFile)
+        {
+            sprintf(section, "TAULA_PRINCIPAL:%s",
+                    pMMBDXP->pField[nIField].FieldName);
+
+            // MM_DEF_LANGUAGE
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_DEF_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_DEF_LANGUAGE] =
+                    '\0';
+
+            // MM_ENG_LANGUAGE
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_eng");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_ENG_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
```

Comment:
The logic seems inverted to me. Currently it is "if the field description for the default language is already set, then copy the value of descriptor_eng in it". I would expect the condition to be "== '\0'" instead. or am I missing something ? If so, a code comment would be helpful

Reply:
I have to review that. I did it yesterday late and some stuff must be reviewed. 
I leave this opened

Reply:
I think you were right. The default one is filled with the idiomatic. Thanks